sips/sip-011/sip-011-catastrophic-failure-recovery.md

```diff
@@ -0,0 +1,464 @@
+must be defined the `services` bitfield in the p2p message preamble that
+distinguishes nodes with the fix applied from nodes that do not have the fix.
+This, in turn, will allow the fixed nodes to know when it is appropriate to use
+the new message types.
+2. The Foundation and other ecosystem entities will deploy new versions of their
+nodes, which use the new `services` bitfield.
+3. Once the Foundation believes that there are enough publicly-routable nodes
+with the fix applied, it will begin executing the critical bugfix rollout
+procedure per above.
+
+In this procedure, the network versioning information is preserved.  Therefore,
+this procedure is only appropriate if backwards-compatibility with other nodes
+and downstream clients can be preserved.
+
+This procedure should be executed instead of changing the networking protocol if
+at all possible.
+
+### Changing Existing Network Messages
+
+If the catastrophic failure can only be remedied by changing the way nodes
+handle an existing message, such that the change renders the node incompatible
+with the network (i.e. unable to effectively participate in block
+synchronization, relaying, and/or mining), then the bugfix will require rolling
+out two concurrent versions of the network that share the same chainstate for a
+time in order to allow people running old nodes a chance to upgrade.  The
+deadline to upgrade (and the deadline at which the old, buggy code-paths will
+cease to be supported) will be enforced programmatically by new nodes, and will
+be measured in burnchain block heights.
+
+This bugfix procedure consists of the following steps.
+
+1. If at all possible, a version of the node software with the buggy code paths
+disabled will be released in order to provide partially-degraded service while
+booting up the new, incompatible version of the network with the fix applied.
+This release may have certain features intentionally disabled, or may introduce
+performance regressions, as long as doing so does not introduce any other
+liveness or safety failures.
+2. The Foundation and other ecosystem entities will run versions of the degraded
+but functional node software from step 1, if applicable.  If a pre-built binary
+needs to be generated to encourage its mass deployment, then it will be
+announced via a signed email from announce@stacks.org as a critical bug-fix
+point release, with instructions on how to proceed in the coming network
+upgrade.
+3. Once the immediate liveness or safety bug no longer manifests (even if it
+means operating the blockchain in a degraded state), the Foundation will procure
+a _testnet_ release of the node software with the incompatible fix applied.
+Power users and ecosystem entities are encouraged to run this testnet release in
+order to verify that it solves the problem and works correctly.
+4. Once the Foundation and at least one other significant ecosystem entity are
```

Comment:
Is it worth qualifying "significant ecosystem entity" for this document? Perhaps "entities operating one or more publicly routable nodes" or "entities operating default bootstrap nodes"?

Reply:
I'm going to remove that term, since it's so hard to qualify.

Reply:
Addressed in 3832ae2.  Since this section pertains to how we're going to repair a broken network, I think it's safe to say that the list of folks who need to sign off on (or at least not actively oppose) a solution to the breakage are simply the folks who showed up in step 2 to help tide the system over from the broken and degraded network to the new network.  These are exactly the folks who will be well-positioned to make the call as to whether or not the new incompatible network has fixed the problem.